### DISCUSSION TITLE
queryCache.invalidateQueries() does not refetch ?

### DISCUSSION THREAD

**Discussion by liorchamla** [posva/pinia-colada#135] [Questions] [answered]
First : I'm sorry I can't give you all a reproduction environment. My question is very simple : 
1. I have a list of users which queries users with pagination, so the key is like that : `() => ["users", route.params.page]` ;
2. I have a view for a single user, and inside, I have a mutation for editing or deleting the user ;
3. After editing or deleting the user, I want to invalidate queryCache in order to make pinia colada refetch the list, so when I go back to the list, it will refetch users, so after mutations are done, I do :  `queryCache.invalidateQueries({ key: ["users"] })`

**Expected behavior**
When I go back to the list, I expect my list to refetch because the cache is no longer valid

**Real behavior**
The list still shows me the old data and does not refetch what so ever

Is there something I'm missing ?

**Comment by posva** [comment]
Yes, [it should refetch](https://pinia-colada.esm.dev/guide/query-invalidation.html#Invalidation-via-Cache-Store). If you can't provide a repro, a failing test case would also work!

**Reply by liorchamla** [reply, marked as answer]
Ok, I got this, the fact that the useQuery was inside a Pinia store was problematic (I can't understand why) but moving the useQuery inside the Nuxt page code made it all work again.

I can understand how not providing any testing environment is hard for you to answer. Anyway, this made the trick !

**Reply by posva** [reply]
I'm glad you found the solution!
BTW `useQuery()` within a store is not problematic per se, it makes the query always active because the store is never _disposed of_.
